Assets/HoloToolkit/UX/Scripts/BoundingBoxes/BoundingBox.cs

```diff
@@ -0,0 +1,482 @@
+                    break;
+
+                case FlattenModeEnum.FlattenY:
+                    newFlattenedAxis = FlattenModeEnum.FlattenY;
+                    targetBoundsLocalScale.y = flattenedAxisThickness * maxAxisThickness;
+                    break;
+
+                case FlattenModeEnum.FlattenZ:
+                    newFlattenedAxis = FlattenModeEnum.FlattenZ;
+                    targetBoundsLocalScale.z = flattenedAxisThickness * maxAxisThickness;
+                    break;
+            }
+
+            FlattenedAxis = newFlattenedAxis;
+        }
+
+        /// <summary>
+        /// Updates bounding box to match target position etc
+        /// </summary>
+        protected virtual void UpdateScaleTransform()
+        {
+            // If we don't have a target, nothing to do here
+            if (target == null)
+                return;
+
+            // Get position of object based on renderers
+            transform.position = targetBoundsWorldCenter;
+            Vector3 scale = targetBoundsLocalScale;
+
+            // Use absolute value when determining smallest axis
+            // so we don't get fooled by inverted scales
+            float largestDimension = Mathf.Max(Mathf.Max(
+                Mathf.Abs(scale.x),
+                Mathf.Abs(scale.y)),
+                Mathf.Abs(scale.z));
+
+            switch (flattenedAxis)
+            {
+                case BoundingBox.FlattenModeEnum.DoNotFlatten:
+                default:
+                    scale.x += (largestDimension * scalePadding);
+                    scale.y += (largestDimension * scalePadding);
+                    scale.z += (largestDimension * scalePadding);
+                    break;
+
+                case BoundingBox.FlattenModeEnum.FlattenX:
+                    scale.x += (largestDimension * flattenedScalePadding);
+                    scale.y += (largestDimension * scalePadding);
+                    scale.z += (largestDimension * scalePadding);
+                    break;
+
+                case BoundingBox.FlattenModeEnum.FlattenY:
+                    scale.x += (largestDimension * scalePadding);
+                    scale.y += (largestDimension * flattenedScalePadding);
+                    scale.z += (largestDimension * scalePadding);
+                    break;
+
+                case BoundingBox.FlattenModeEnum.FlattenZ:
+                    scale.x += (largestDimension * scalePadding);
+                    scale.y += (largestDimension * scalePadding);
+                    scale.z += (largestDimension * flattenedScalePadding);
+                    break;
+            }
+
+
+            scaleTransform.localScale = scale;
+
+            Vector3 rotation = target.transform.eulerAngles;
+
+            transform.eulerAngles = rotation;
+        }
+
+        [Header("Objects")]
+        [Tooltip("The target object")]
+        [SerializeField]
+        protected GameObject target;
```

Comment:
nit: move all fields and properties to the top of the class

Reply:
Moved.